qcodes/instrument/base.py

```diff
@@ -183,10 +201,10 @@ def snapshot_base(self, update: bool=False,
             except:
                 # really log this twice. Once verbose for the UI and once
                 # at lower level with more info for file based loggers
-                log.warning(f"Snapshot: Could not update parameter: "
-                            f"{name} on {self.full_name}")
-                log.info(f"Details for Snapshot of {name}:",
-                         exec_info=True)
+                self.log.warning(f"Snapshot: Could not update parameter: "
+                                 f"{name}")
+                self.log.info(f"Details for Snapshot:",
```

Comment:
yes, there is unfortunately no other way. I could go through the instruments and add those lines. For now all the log messages from the `BaseInstrument` get logged.

Reply:
We should really have a how to write an instrument guide where things like this are mentioned. For now it might be good to add it to https://github.com/QCoDeS/Qcodes/blob/master/docs/examples/Creating%20Instrument%20Drivers.ipynb

Reply:
Fully agreed. I can add it there.